src/runtime_intrinsics.c

```diff
@@ -390,21 +398,27 @@ static inline jl_value_t *jl_intrinsic_cvt(jl_value_t *ty, jl_value_t *a, const
 {
     jl_ptls_t ptls = jl_get_ptls_states();
     jl_value_t *aty = jl_typeof(a);
-    if (!jl_is_bitstype(aty))
-        jl_errorf("%s: value is not a bitstype", name);
-    if (!jl_is_bitstype(ty))
-        jl_errorf("%s: type is not a bitstype", name);
+    if (!(jl_is_bitstype(aty) || jl_is_vec_type(aty)))
```

Comment:
I'm wondering if this idiom of getting jl_typeof and then running the two checks is worth factoring out.